spec/controllers/projects_controller_spec.rb

```diff
@@ -193,6 +193,22 @@
               expect(assigns[:story].project).to eq(project)
             end
           end
+
+          describe 'when the user change to another project from another team' do
+            let(:new_team)            { create :team }
+            let(:new_project)         { create :project, users: [user] }
+            let(:new_story)           { create :story, project: new_project, requested_by: user }
+
+            it 'should accept request when it is from registred team', :aggregate_failures do
+              new_team.projects << new_project
+              user.teams << new_team
+
+              get :show, params: { id: new_project }
```

Comment:
We should have a test where we do a get to one team, then to other team and see if the selected one is the last one too.